### ISSUE TITLE
Maximum call stack size exceeded because of accessing order in redefinition

### ISSUE THREAD

**Issue by vonagam** [stalniy/bdd-lazy-var#20]
So this will fail:

```js
describe( 'parent', () => {
  def( 'value', () => 1 );

  describe( 'child', () => {
    def( 'addition', () => 1 );
    def( 'value', () => $addition + $value );

    it( 'is two', () => expect( $value ).to.eq( 2 ) );
  } );
} );
```

If we change second `value` definition from `$addition + $value` to `$value + $addition` it will pass.

I don't know if this is a bug or this is how it is supposed to be. Feel free to close if this is not a problem. ( But then maybe it is good to have a note about this somewhere )

**Comment by stalniy** [comment]
Thanks for the issue. It's definitely a bug. I will check what is wrong there. Thanks for the report!

**Comment by stalniy** [comment]
fixed and published 1.3.0

**Comment by vonagam** [comment]
Thanks for the fix and for the library.
